packages/core/src/RushTool.ts

```diff
@@ -0,0 +1,64 @@
+import path from "path";
+import fs from "fs-extra";
+import jju from "jju";
+
+import { PackageJSON } from "@changesets/types";
+import { Tool, ToolType, Package, Packages, MonorepoRoot } from './Tool';
+
+interface RushJson {
+    projects: RushProject[];
+}
+
+interface RushProject {
+    packageName: string;
+    projectFolder: string;
+}
+
+export const RushTool: Tool = {
+    type(): ToolType {
+        return 'rush';
+    },
+
+    async isMonorepoRoot(directory: string): Promise<MonorepoRoot | undefined> {
+        try {
+            await fs.readFile(path.join(directory, "rush.json"), "utf8");
+            return {
+                dir: directory,
+                tool: RushTool
+            };
+        } catch (err) {
+            if (err.code !== "ENOENT") {
+                throw err;
+            }
+        }
+    },
+
+    async getPackages(directory: string): Promise<Packages | undefined> {
+        try {
+            const rushText: string = await fs.readFile(path.join(directory, "rush.json"), "utf8");
+
+            // Rush configuration files are full of inline and block-scope comment blocks (JSONC),
+            // so we use a parser that can handle that.
+            const rushJson: RushJson = jju.parse(rushText);
+
+            const directories = rushJson.projects.map(project => path.resolve(directory, project.projectFolder)).sort();
+            const packageJsons = await Promise.all(directories.map(dir => fs.readJson(path.join(dir, "package.json"))));
+            const packages = directories.map((dir, index) => {
+              return {
+                dir,
+                packageJson: packageJsons[index] as PackageJSON
+              };
+            });
```

Comment:
nit: I think that it would be slightly more readable (no index-based zipping) if we'd merge this with the `Promise.all` "loop" a line above